src/Type/Php/SprintfFunctionDynamicReturnTypeExtension.php

```diff
@@ -115,19 +119,19 @@ public function getTypeFromFunctionCall(
 			$returnType = new StringType();
 		}
 
-		return $this->getConstantType($args, $returnType, $functionReflection, $scope);
```

Comment:
instead of building the general return type first and pass it as a default-return into `getConstantType()` we now invoke `getConstantType()` first and build the general return type only if we were not able to infer a constant type in the first place